scripts/wrappers/reset.py

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/python3
+import click
+import os
+import subprocess
+import sys
+import time
+
+from common.utils import (
+    get_available_addons,
+    get_current_arch,
+    get_status,
+    snap_data,
+    wait_for_ready,
+    snap_common,
+    is_cluster_locked,
+    exit_if_no_permission,
+    ensure_started,
+    kubeconfig,
+)
+
+
+def exit_if_multinode():
+    """
+    Exit if we cannot get the list of nodes or if we are in a multinode cluster
+    """
+    cmd = f"{os.environ['SNAP']}/kubectl {kubeconfig} get no -o name"
+    res = run_silently(cmd)
+    if not res:
+        print("Failed to query the cluster nodes.")
+        sys.exit(1)
+    nodes = res.split()
+    if len(nodes) > 1:
+        print(
+            "This is a multi-node MicroK8s deployment. Reset is applicable for single node clusters."
+        )
+        print("Please remove all joined nodes before calling reset.")
+        sys.exit(0)
+
+
+def disable_addon(repo, addon, args=[]):
+    """
+    Try to disable an addon. Ignore any errors and/or silence any output.
+    """
+    wait_for_ready(timeout=30)
+    subprocess.run(
+        [snap_common() / "addons" / repo / "addons" / addon / "disable", *args],
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+    )
+    wait_for_ready(timeout=30)
+
+
+def disable_addons(destroy_storage):
+    """
+    Iterate over all addons and disable the enabled ones.
+    """
+    print("Disabling all addons")
+
+    available_addons_info = get_available_addons(get_current_arch())
+    enabled, disabled = get_status(available_addons_info, True)
+    for addon in available_addons_info:
+        # Do not disable HA
+        if addon["name"] == "ha-cluster":
+            continue
+
+        print(f"Disabling addon : {addon['repository']}/{addon['name']}")
+        # Do not disable disabled addons
+        if addon in disabled:
+            continue
```

Comment:
```suggestion
    enabled, _ = get_status(available_addons_info, True)
    for addon in enabled:
        # Do not disable HA
        if addon["name"] == "ha-cluster":
            continue

        print(f"Disabling addon : {addon['repository']}/{addon['name']}")

Reply:
Thank you for the suggestion. This slightly changes the output by printing only the enabled addons. Let's not do this change for now.